protocols/kad/src/handler.rs

```diff
@@ -573,24 +547,41 @@ impl Handler {
             });
     }
 
-    fn on_dial_upgrade_error(
-        &mut self,
-        DialUpgradeError {
-            info: (), error, ..
-        }: DialUpgradeError<
-            <Self as ConnectionHandler>::OutboundOpenInfo,
-            <Self as ConnectionHandler>::OutboundProtocol,
-        >,
-    ) {
-        // TODO: cache the fact that the remote doesn't support kademlia at all, so that we don't
-        //       continue trying
+    fn request_new_stream(&mut self, id: QueryId, msg: KadRequestMsg) {
```

Comment:
With the fact that this method does not emit a `ConnectionHandlerEvent::OutboundSubstreamRequest`, I find the name missleading. Can you think of an alternative?